llvm/lib/Target/RISCV/RISCVInstrInfo.cpp

```diff
@@ -2898,13 +2898,13 @@ bool RISCVInstrInfo::verifyInstruction(const MachineInstr &MI,
           Ok = isShiftedUInt<4, 1>(Imm);
           break;
         case RISCVOp::OPERAND_UIMM5_NONZERO:
-          Ok = isUInt<5>(Imm) && (Imm != 0);
+          Ok = Imm >= 1 && Imm <= 31;
```

Comment:
The original code was a direct translation of the name. So that seems a little clearer to me.

Reply:
Checking the lower bound once is easier to read for me.
But that's not a strong preference. How about I revert the `isXXX && oneCond` cases to leave only the `plus1`/`minus1` changed?

Reply:
personally I also think that using isUInt<N> instead of "magic" constants is easier to read.

Reply:
> How about I revert the isXXX && oneCond cases to leave only the plus1/minus1 changed?

That works for me.